### PR TITLE
JDK-8264454 : Jaxp unit test from open jdk needs to be improved

### PR DESCRIPTION
**Pull request by mahendrachhipa**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264454](https://bugs.openjdk.java.net/browse/JDK-8264454): Jaxp unit test from open jdk needs to be improved

### Reviewers
 * [Joe Wang](https://openjdk.java.net/census#joehw) (@JoeWang-Java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3274/head:pull/3274` \
`$ git checkout pull/3274`

Update a local copy of the PR: \
`$ git checkout pull/3274` \
`$ git pull https://git.openjdk.java.net/jdk pull/3274/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3274`

View PR using the GUI difftool: \
`$ git pr show -t 3274`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3274.diff">https://git.openjdk.java.net/jdk/pull/3274.diff</a>

</details>

### CODE REVIEW

File: test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java [hunk 1]
```diff
@@ -158,44 +170,40 @@ public final void testNewDurationYearMonthLexicalRepresentation1() {
...
-                try {
-                    QName xmlSchemaType = duration.getXMLSchemaType();
-                    if (!xmlSchemaType.equals(DatatypeConstants.DURATION_YEARMONTH)) {
-                        Assert.fail("Duration created with XMLSchemaType of\"" + xmlSchemaType + "\" was expected to be \""
+            // right XMLSchemaType?
+            // TODO: enable test, it should pass, it fails with Exception(s)
+            // for now due to a bug
+            try {
+                QName xmlSchemaType = duration.getXMLSchemaType();
+                if (!xmlSchemaType.equals(DatatypeConstants.DURATION_YEARMONTH)) {
```
Comment (JoeWang-Java) [inline test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java:194]: Can be changed to assertEquals

File: test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java [hunk 2]
```diff
@@ -158,44 +170,40 @@ public final void testNewDurationYearMonthLexicalRepresentation1() {
...
+                    Assert.fail("Duration created with XMLSchemaType of\"" + xmlSchemaType + "\" was expected to be \""
                                 + DatatypeConstants.DURATION_YEARMONTH + "\" and has the value \"" + duration.toString() + "\"");
-                    }
+                }
                 } catch (IllegalStateException illegalStateException) {
                     // TODO; this test really should pass
                     System.err.println("Please fix this bug that is being ignored, for now: " + illegalStateException.getMessage());
```
Comment (JoeWang-Java) [inline test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java:200]: Do we still have such a bug?

File: test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java [hunk 3]
```diff
@@ -158,44 +170,40 @@ public final void testNewDurationYearMonthLexicalRepresentation1() {
...
                 }
 
                 // does it have the right value?
-                if (!TEST_VALUES_LEXICAL[onTestValue + 1].equals(duration.toString())) {
-                    Assert.fail("Duration created with \"" + TEST_VALUES_LEXICAL[onTestValue] + "\" was expected to be \""
-                            + TEST_VALUES_LEXICAL[onTestValue + 1] + "\" and has the value \"" + duration.toString() + "\"");
+                if (!expectedLex.equals(duration.toString())) {
```
[the remark this answers is not in the document]
Comment (mahendrachhipa) [reply test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java:204]: done

File: test/jaxp/javax/xml/jaxp/unittest/common/Bug6350682.java [hunk 4]
```diff
@@ -46,32 +44,20 @@
...
-            if (Bug6350682.class.getClassLoader() == null)
-                System.out.println("this class loader is NULL");
-            else
-                System.out.println("this class loader is NOT NULL");
-            SAXParserFactory factory = SAXParserFactory.newInstance();
-            Assert.assertTrue(factory != null, "Failed to get an instance of a SAXParserFactory");
-        } catch (Exception e) {
-            e.printStackTrace();
-            Assert.fail("Exception occured: " + e.getMessage());
+        runWithAllPerm(() -> Thread.currentThread().setContextClassLoader(null));
```
[the remark this answers is not in the document]
Comment (mahendrachhipa) [reply test/jaxp/javax/xml/jaxp/unittest/common/Bug6350682.java:47]: Added the comment

File: test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java (hunk 2 above)
Comment (mahendrachhipa) [reply test/jaxp/javax/xml/jaxp/unittest/datatype/Bug6937964Test.java:200]: Yes, test is still trowing IllegalStateException.

**Comment by JoeWang-Java** [comment]
Ok, this is enough of improvement for now. I created a bug to keep track of the ISE case.

**Review by JoeWang-Java — APPROVED** [review]

**Pull request closed without merging**